### PR TITLE
Refactor azorius contracts

### LINKED ISSUE #952: Update frontend for new Usul contract
### Overview

The frontend needs to be updated to support the new modified Usul contract. 
Changes affecting the frontend include:

**LinearTokenVoting**
- `OZLinearVoting` contract renamed to `LinearTokenVoting` 
- `receiveStrategy` function renamed to `timelockProposal`
- New `isProposer` function that returns whether an address can submit a proposal. This probably won't need to be implemented on the FE yet since this implementation always returns True
- `getProposal` function added for getting proposal details
- `calculateWeight` renamed to `getVotingWeight`
- `quorumDenominator` constant updated from 100 to 1,000,000

**FractalUsul**
- `submitProposalWithMetadata` function renamed to `submitProposal`
- `submitProposal` interface changed to:
    `function submitProposal(
        address strategy,
        bytes memory data,
        FractalUsul.Transaction[] calldata transactions,
        string calldata metadata
    ) external;`
- `ProposalCreated` event changed to: 
`    event ProposalCreated(
        address strategy,
        uint256 proposalId,
        address proposer,
        Transaction[] transactions,
        string metadata
    );`
- This function now emits a single event, called `ProposalCreated`. This event had a duplicate of proposald / proposalNumber, so this duplicate was also removed. This event now includes the following fields:
`    event ProposalCreated(
        address strategy,
        uint256 proposalId,
        address proposer,
        Transaction[] transactions,
        string title,
        string description, 
        string documentationUrl
    );`
- `Executing` state renamed to `Executable`
- `StrategyFinalized` event renamed to `ProposalTimelocked`
- `receiveStrategy` function renamed to `timelockProposal`
- Proposal states `Uninitialized` and `Cancelled` removed
- `state` function renamed to `proposalState` for getting the state of a proposal
- ProposalState Enum updated to: 
    `enum ProposalState {
        ACTIVE,
        TIMELOCKED,
        EXECUTABLE,
        EXECUTED
    }`

**UsulVetoGuard & VetoGuard**
- `queueProposal` function renamed to `timelockProposal`
- `ProposalQueued` event renamed to `ProposalTimelocked`

### Link to documentation

_No response_

### PR DESCRIPTION
**Pull request by Da-Colon**
## Description
Hold on to your butts. Azorius Refactor incoming. Combined effort from the entire engineering team. Refactors and updates to newly released `fractal-contracts`

## Notes
- new block timestamp methods `getTimeStamp` and `getEstimatedNumberOfBlocks` to help with converting minutes to blocks per updates to using blocks instead of timestamps in the contracts

## Issue / Notion doc (if applicable)
Closes #952

## Testing
Regression testing! Test anything and everything but if you forget here is a short list:

### Parent DAOs
- create Multisig Parent DAO
- create Azorius Parent DAO
- Delegate Modal 
- Create Azorius Parent (TokenWrapped) DAO
- Wrap Token Modal
- Unwrap Token Modal
- Create Proposal (Multisig Parent)
- Create Proposal (Azorius Parent)
- Create Proposal (Azorius Parent (TokenWrapped))
- Send Assets Modal

### Child DAOs
- view Heirarchy
- create Multisig with Multisig Parent DAO 
- create Multisig with Azorius Parent DAO
- Create Multisig with Azorius Parent (TokenWrapped) DAO
- create Azorius with Azorius Parent DAO
- create Azorius with Multisig Parent DAO
- Create Azorius with Azorius Parent (TokenWrapped) DAO
- Create Azorius (TokenWrapped) with Azorius Parent (TokenWrapped) DAO
- Delegate Modal 
- Create Proposal (Multisig Child)
- Create Proposal (Azorius Child)
- Create Proposal (Azorius Child (TokenWrapped))
- Freeze DAO (Multisig Child)
- Freeze DAO (Azorius Child)
- Freeze DAO (Azorius Child (TokenWrapped))
- Send Assets Modal

### CODE REVIEW

**Review by tbwebb22 — COMMENTED** [review]
When creating a child token DAO, I'm seeing an execution period field twice, once on the 'Compose Governance' page, and once on the 'Parent Governance' page

**Review by herbig — CHANGES_REQUESTED** [review]
Hey @Da-Colon I've tested everything except for Freeze and the Azorius proposal lifecycle, here's the major stuff:

- [ ] DAO name doesn't always appear after deploying a new DAO without a refresh (this is definitely preexisting, and explains why our tests are often failing)
- [ ] helperFreezeVotesThreshold doesn't show in both multisig / Azorius subDAO creation (from a multisig parent, Azorius parent works)
- [x] Create proposal doesn't work on a multisig DAO, button enables but clicking it doesn't open Metamask, notification briefly pops up. It works fine on Token voting however.
- [ ] Azorius new proposals don't ever appear in the proposals list (can't test executing them currently)

Once we have proposals list for Azorius displaying I'll go ahead and test proposal states as well.

And is there an easy way to find a subDAO without the DAO hierarchy?

File: src/hooks/safe/useSafeContracts.ts [hunk 1]
```diff
@@ -33,11 +33,11 @@ export default function useSafeContracts() {
...
-      vetoMultisigVotingMasterCopy,
-      vetoERC20VotingMasterCopy,
-      votesTokenMasterCopy,
+      multisigFreezeGuardMasterCopy,
+      azoriusFreezeGuardMasterCopy,
+      multisigFreezeVotingMasterCopy,
+      erc20FreezeVotingMasterCopy,
+      votesERC20MasterCopy,
       claimingMasterCopy,
       votesERC20WrapperMasterCopy,
```
[the remark this answers is not in the document]
Comment (Da-Colon) [reply src/hooks/safe/useSafeContracts.ts:42]: done and done

File: src/hooks/DAO/useDeployDAO.ts [hunk 2]
```diff
@@ -16,7 +16,7 @@ const useDeployDAO = () => {
   const { t } = useTranslation('transaction');
 
   const deployDao = useCallback(
-    (daoData: GnosisDAO | TokenGovernanceDAO, successCallback: (daoAddress: string) => void) => {
+    (daoData: GnosisDAO | AzoriusGovernanceDAO, successCallback: (daoAddress: string) => void) => {
```
[the remark this answers is not in the document]
Comment (Da-Colon) [reply src/hooks/DAO/useDeployDAO.ts:19]: GnosisDAO -> SafeMultisigDAO. I started to do a full search for gnosis. but might need a little more of a delicate search (seperate PR).

File: src/types/transaction.ts [hunk 3]
```diff
@@ -15,6 +15,15 @@ export interface MetaTransaction {
   operation: number;
 }
 
+export interface MetaData {
```
[the remark this answers is not in the document]
Comment (Da-Colon) [reply src/types/transaction.ts:18]: There is actually another type called `ProposalMetaData`. it has similar but different properties. Can probably do a type clean up later

File: src/models/FreezeGuardTxBuilder.ts [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+      'setUp',
+      [
+        ethers.utils.defaultAbiCoder.encode(
+          ['address', 'uint256', 'uint256', 'uint256', 'uint256', 'address', 'address'],
+          [
+            this.parentAddress, // Owner -- Parent DAO
+            subDaoData.freezeVotesThreshold, // FreezeVotesThreshold
+            subDaoData.freezeProposalPeriod, // FreezeProposalPeriod
+            subDaoData.freezePeriod, // FreezePeriod
+            this.parentTokenAddress ?? this.parentAddress, // ParentGnosisSafe or Votes Token
```
Comment (herbig) [inline src/models/FreezeGuardTxBuilder.ts:107]: @tbwebb22 does this look good for freeze votes (multisig / azorius) setUp functions?

Comment (tbwebb22) [reply src/models/FreezeGuardTxBuilder.ts:107]: Yeah I think everything here looks good. Had to think a bit about that last address param line, but it should work. If the parent DAO does have a token, then we will be deploying the ERC20FreezeVoting and passing to it the token address. If the parent DAO doesn't have a token, then we will be deploying the MultisigFreezeVoting and passing the parent DAO address.

**Comment by Da-Colon** [comment]
boom, fixed the create proposal by using `useAsyncRetry` to poll `getProposalVotes` until the tx has been mined. 

Based on the fix for `useAsyncRetry` This may have been related to the tests failing as well?

File: src/components/pages/DaoDashboard/Activities/ActivityFreeze.tsx [hunk 5]
```diff
@@ -22,10 +22,10 @@ export function FreezeDescription({ isFrozen }: { isFrozen: boolean }) {
 
 export function ActivityFreeze({
   freezeGuard,
-  vetoVotingContract,
+  freezeVotingContract: freezeVotingContract,
```
[the remark this answers is not in the document]
Comment (herbig) [reply src/components/pages/DaoDashboard/Activities/ActivityFreeze.tsx:25]: yeah, weird, this is probably a result of doing a rename in VS code, I didn't intend to add that here

File: src/components/Proposals/ProposalsList.tsx [hunk 6]
```diff
@@ -16,7 +16,7 @@ export function ProposalsList({ proposals }: { proposals: FractalProposal[] }) {
   } = useFractal();
 
   const showCreateButton =
-    type === StrategyType.GNOSIS_SAFE_AZORIUS ? true : safe?.owners.includes(user.address || '');
+    type === StrategyType.AZORIUS ? true : safe?.owners.includes(user.address || '');
```
[the remark this answers is not in the document]
Comment (herbig) [reply src/components/Proposals/ProposalsList.tsx:19]: yeah agreed, we have strategy types in Azorius, and these are governance modules

File: src/types/createDAO.ts [hunk 7]
```diff
@@ -6,10 +6,10 @@ import { StrategyType } from './fractal';
 import { EthAddress } from './utils';
 export enum CreatorSteps {
   ESSENTIALS = 'essentials',
-  GNOSIS_GOVERNANCE = 'gnosis',
-  GNOSIS_WITH_AZORIUS = 'govToken',
+  MULTISIG_GOVERNANCE = 'gnosis',
```
[the remark this answers is not in the document]
Comment (herbig) [reply src/types/createDAO.ts:9]: @mudrila it doesn't look like these enum values are necessary for anything, they're not string translation keys, so we should be able to just remove them right?

File: src/types/createDAO.ts [hunk 8]
```diff
@@ -6,10 +6,10 @@ import { StrategyType } from './fractal';
...
+  AZORIUS_GOVERNANCE = 'govToken',
```
[the remark this answers is not in the document]
Comment (herbig) [reply src/types/createDAO.ts:10]: https://github.com/decent-dao/fractal-interface/pull/1152#discussion_r1188604957

**Pull request merged**